src/main/java/core/basesyntax/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    private final Storage storage;
+
+    public SupplyOperation(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void count(FruitTransaction transaction) {
+        if (transaction.getQuantity() < 0) {
+            throw new RuntimeException("Quantity can't be less than 0");
+        }
+        storage.addFruit(transaction.getFruit(), transaction.getQuantity());
+    }

```

Comment:
The `SupplyOperation` class should not have a dependency on `Storage` directly. It would be better to pass the `Storage` as an argument to the `count` method to keep the class aligned with the SOLID principles, specifically the Dependency Inversion Principle. This way, `SupplyOperation` is not directly coupled to a specific storage implementation.